hermite.go

```diff
@@ -0,0 +1,383 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package quad
+
+import (
+	"math"
+	"math/cmplx"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/mathext/airy"
+)
+
+// Hermite integrates a function over the whole real line with negative
+// squared exponential weight
+//  int_-inf^inf e^-x^2 f(x) dx .
+type Hermite struct{}
+
+func (h Hermite) FixedLocations(x, weight []float64, min, max float64) {
+	// TODO(btracey): Implement the case where x > 20, x < 200 so that we don't
+	// need to store all of that data.
+
+	// References:
+	// Algorithm:
+	// G. H. Golub and J. A. Welsch, "Calculation of Gauss quadrature rules",
+	// Math. Comp. 23:221-230, 1969.
+	// A. Glaser, X. Liu and V. Rokhlin, "A fast algorithm for the
+	// calculation of the roots of special functions", SIAM Journal
+	// on Scientific Computing", 29(4):1420-1438:, 2007.
+	// A. Townsend, T. Trogdon and S. Olver, Fast computation of Gauss
+	// nodes and weights on the whole real line, submitted, 2014.
+	//
+	// Algorithm adapted from Chubfun http://www.chebfun.org/.
+
+	if len(x) != len(weight) {
+		panic("hermite: slice length mismatch")
+	}
+	if min >= max {
+		panic("hermite: min >= max")
+	}
+	if !math.IsInf(min, -1) || !math.IsInf(max, 1) {
+		panic("hermite: non-infinite bound")
+	}
+	h.locations(x, weight)
+}
+
+func (h Hermite) locations(x, weights []float64) {
+	n := len(x)
+	switch {
+	default:
```

Comment:
This is strange: with the other two cases there is nothing left for the default case, the slice length will not be negative, and if the length is 0, the index will become -1. Change to:
```
	switch {
	case 0 < n && n <= 200:
		copy(x, xCacheHermite[n-1])
		copy(weights, wCacheHermite[n-1])
	case n > 200:
		xasy, weightsasy := h.locationsAsy(n)
		copy(x, xasy)
		copy(weights, weightsasy)
	}
```
? Also depends on what we want to do when n == 0.